Build external resources cleanups bis

sambuc opened the pull request:
A series of cleanups. It should be easier to review and validate these commits by commits than as a lump package as they are each focused on a single rename / topic.

sambuc commented:
the first commit of this stack is [fix: rename list_servers to list_sessions](https://github.com/SwissDataScienceCenter/renku-data-services/pull/747/commits/a35b473f6c2372c5409442503a34127e31bd2731)

olevski commented on Makefile:
```diff
@@ -169,3 +182,17 @@ amalthea_schema:  ## Updates generates pydantic classes from CRDs
+
+.PHONY: devcontainer_exec
+devcontainer_exec: devcontainer_up
+	devcontainer exec --container-id renku-data-services_devcontainer-data_service-1 -- bash
```
The default is the data services container. You dont have to set it.

```suggestion
	devcontainer exec -- bash
```

sambuc commented:
As discussed I rebased the branch instead of merging in the changes from main. I also added a new commit which updates comments and error string to better represent we have Jupyter and Amalthea sessions.

olevski reviewed (approved):
Thank you for doing this Lionel! It looks great.
